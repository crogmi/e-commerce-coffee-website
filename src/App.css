nav {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  min-height: 5vh;
  background-color: black;
  color: white;
}

.page-title {
  text-align: center;
}