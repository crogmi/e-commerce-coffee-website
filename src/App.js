import React, { useState } from 'react';
import Navbar from './components/Navbar/Navbar';
import Home from './containers/Home/Home';
import About from './containers/AboutPage/AboutPage';
import Order from './containers/OrderPage/OrderPage';
import Product from './containers/Product/Product';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import './App.css';

const App = ({match}) => {

  const ROUTES = {
    HOME: '/',
    ABOUT: '/about',
    ORDER: '/order',
    ORDERID: '/order/:_id'
  }

  const [cart, setCart] = useState([]);

  const addToCart = (name, price, syrup, milk, flavor, temp) => {
    let product = {
      name: name,
      price: price,
      syrup: syrup,
      milk: milk,
      flavor: flavor,
      temp: temp
    };

    setCart((prev) => {
      return [...prev, product]
    });
  };

  return (
    <Router>
      <div className='App'>
        <Navbar />
        <main>
          <Switch>
            <Route path={ROUTES.HOME} exact component={Home} />
            <Route path={ROUTES.ABOUT} component={About} />
            <Route path={ROUTES.ORDER} exact component={Order} />
            <Route path={ROUTES.ORDERID} component={Product}/>
          </Switch>
        </main>
      </div>
    </Router>
  );
};

export default App;
