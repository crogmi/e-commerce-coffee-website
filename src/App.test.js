import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import App from './App';

/* *********************************** Smoke Test *********************************** */

describe('App', () => {
    it('renders without crashing', () => {
        render(<App />)
        screen.debug();
    })


/* *********************************** Snapshot Test *********************************** */

    it('renders correctly', () => {
        const app = render(<App />);
        expect(app).toMatchSnapshot();
    })

    
});