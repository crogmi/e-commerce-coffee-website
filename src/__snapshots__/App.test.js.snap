// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <nav>
          <a
            class="text-links"
            href="/"
          >
            <h3
              class="nav-logo"
            >
              Logo
            </h3>
          </a>
          <ul
            class="nav-links"
          >
            <a
              class="text-links"
              href="/about"
            >
              <li
                class="list-items"
              >
                About
              </li>
            </a>
            <a
              class="text-links"
              href="/order"
            >
              <li
                class="list-items"
              >
                Order
              </li>
            </a>
          </ul>
        </nav>
        <main>
          <div
            class="Home"
          >
            <h1
              class="page-title"
            >
              CoCo Coffee
            </h1>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <nav>
        <a
          class="text-links"
          href="/"
        >
          <h3
            class="nav-logo"
          >
            Logo
          </h3>
        </a>
        <ul
          class="nav-links"
        >
          <a
            class="text-links"
            href="/about"
          >
            <li
              class="list-items"
            >
              About
            </li>
          </a>
          <a
            class="text-links"
            href="/order"
          >
            <li
              class="list-items"
            >
              Order
            </li>
          </a>
        </ul>
      </nav>
      <main>
        <div
          class="Home"
        >
          <h1
            class="page-title"
          >
            CoCo Coffee
          </h1>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
